examples/src/main/java/io/grpc/examples/routeguide/RouteGuideClient.java

```diff
@@ -271,10 +269,40 @@ public static void main(String[] args) throws InterruptedException {
     }
   }
 
-  private static void info(String msg, Object... params) {
+  /**
+   * Make it visible for unit test, so that test can verify state changes by spying on it.
+   */
+  @VisibleForTesting
+  void info(String msg, Object... params) {
     logger.log(Level.INFO, msg, params);
   }
 
+  /**
+   * Make it visible for unit test, so that test can verify state changes by spying on it.
+   */
+  @VisibleForTesting
+  void warning(String msg, Object... params) {
+    logger.log(Level.WARNING, msg, params);
+  }
+
+  /**
+   * Make it visible for unit test, so that we can override it by spying on it, as we do not want to
+   * introduce randomness in unit test.
+   */
+  @VisibleForTesting
+  int randomIndex(int bound, Random random) {
```

Comment:
The correct way to do this is make a setter for the Random instance and attach it to this class.